### ISSUE TITLE
Environment variables ignored if the value provided in the configuration file 

### ISSUE THREAD

**Issue by AlexSkrypnyk** [drush-ops/drush#5960]
**Describe the bug**
Given `$HOME/.drush/drush.yml` file with the following content:
```
options:
  root: myroot
  uri: mysite.local
```
When I call

```
DRUSH_OPTIONS_URI=other.domain ./vendor/bin/drush st
```
Then I see the output
```
Drupal version : 10.1.4                                      
Site URI       : mysite.local
```

**Expected behavior**
Then I see the output
```
Drupal version : 10.1.4                                      
Site URI       : other.domain
```

**Workaround**
Is there another way to do the desired action?

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 12.5.11
| Drupal version? | 10.1
| PHP version     | 8.1
| OS?             | Mac

**Additional information**

The issue lies in Drush `ConfigLocator`. The constructor adds the user context after env context. `consolidation/log`'s `ConfigOverlay::findContext()` looks up the contexts in reversed order hence the `options.uri` defined in `/home/.drush/drush.yml` (user context) supersedes the `DRUSH_OPTIONS_URI` var (env context).

In `ConfigLocator::__contruct()`
```
$this->config->addPlaceholder(self::DRUSH_CONTEXT);
if (!empty($envPrefix)) {
    $envConfig = new EnvConfig($envPrefix);
    $this->config->addContext(self::ENV_CONTEXT, $envConfig);
}
$this->config->addPlaceholder(self::USER_CONTEXT);
```
If I move `USER_CONTEXT` above `ENV_CONTEXT` the `DRUSH_OPTIONS_URI` will take precedence as expected.

**Comment by weitzman** [comment]
Fixed in #5961

**Comment by AlexSkrypnyk** [comment]
This will be available in Drush 13. Not backported to Drush 12.
